apps/module-presale/contracts/Presale.sol

```diff
@@ -0,0 +1,336 @@
+        Funding,     // Sale has started and contributors can purchase tokens.
+        Refunding,   // Sale did not reach daiFundingGoal within fundingPeriod and contributors may claim refunds.
+        GoalReached, // Sale reached daiFundingGoal and the Fundraising app is ready to be initialized.
+        Closed       // After GoalReached, sale was closed and the Fundraising app was initialized.
+    }
+
+    /*
+     * Initialization
+     */
+
+    /**
+    * @notice Initialize Presale app with `_daiToken` to be used for purchasing `_projectToken`, controlled by `_projectTokenManager`. Project tokens are provided in vested form using `_vestingCliffPeriod` and `_vestingCompletePeriod`. The Presale accepts dai until `_daiFundingGoal` is reached. `percentSupplyOffered` is used to calculate the dai to project token exchange rate. The presale allows project token purchases for `_fundingPeriod` after the sale is started. If the funding goal is reached, part of the raised funds are sent to `_fundraisingPool`, associated with a Fundraising app. The raised funds that are not sent to the fundraising pool are sent to `_beneficiaryAddress` according to the ratio specified in `_percentFundingForBenefiriary`.
+    * @param _daiToken ERC20 Token accepted for purchasing project tokens.
+    * @param _projectToken MiniMeToken project tokens being offered for sale in vested form.
+    * @param _projectTokenManager TokenManager Token manager in control of the offered project tokens.
+    * @param _vestingCliffPeriod uint64 Cliff period used for vested project tokens.
+    * @param _vestingCompletePeriod uint64 Complete period used for vested project tokens.
+    * @param _daiFundingGoal uint256 Target dai funding goal.
+    * @param _percentSupplyOffered uin256 Percent of the total supply of project tokens that will be offered in this sale and in further fundraising stages.
+    * @param _fundingPeriod uint64 The period within which this sale accepts project token purchases.
+    * @param _fundraisingPool Pool The fundraising pool associated with the Fundraising app where part of the raised dai tokens will be sent to, if this sale is succesful.
+    * @param _beneficiaryAddress address The address to which part of the raised dai tokens will be sent to, if this sale is successful.
+    * @param _percentFundingForBenefiriary uint256 The percentage of the raised dai tokens that will be sent to the beneficiary address, instead of the fundraising pool, when this sale is closed.
+    */
+    function initialize(
+        ERC20 _daiToken,
+        MiniMeToken _projectToken,
+        TokenManager _projectTokenManager,
+        uint64 _vestingCliffPeriod,
+        uint64 _vestingCompletePeriod,
+        uint256 _daiFundingGoal,
+        uint256 _percentSupplyOffered,
+        uint64 _fundingPeriod,
+        address _fundraisingPool,
+        address _beneficiaryAddress,
+        uint256 _percentFundingForBenefiriary
```

Comment:
Perhaps it would be useful to allow a user to initialize the app with a pre-determined start date, and offer functionality to change that start date (rather than asking them to send a tx to start the fundraising)?

I imagine most fundraising instances would happen on pre-determined, well-communicated timestamps.

Reply:
I like this, it does seem like it would be nice to be able to plan the start time and would make communications easier. 👍

Reply:
Yep, this sounds like a very good idea. Confirming with @osarrouy and @lkngtn

Reply:
Yep. That sounds good to me: a more trustless setup :)

Reply:
Ok, I'll implement the change, but not right away, since it's not trivial.

Reply:
It's implemented. Unfortunately, we've lost the SaleStarted event, but I believe that the value of this feature is worth it.